mm2src/mm2_metamask/src/eth_provider.rs

```diff
@@ -0,0 +1,168 @@
+//! This module is inspired by https://github.com/tomusdrw/rust-web3/blob/master/src/transports/eip_1193.rs
```

Comment:
For the next iteration: should we update `rust-web3` dependency instead? We use a quite outdated fork now.

Reply:
This can be difficult due to the fact that [Eip1193 transport](https://github.com/tomusdrw/rust-web3/blob/master/src/transports/eip_1193.rs#L157-L158) is local only, so we'll need to make a workaround anyway.
Currently, `EthProvider` is Send because it's wrapped into the event-driven pattern

Reply:
> This can be difficult due to the fact that [Eip1193 transport](https://github.com/tomusdrw/rust-web3/blob/master/src/transports/eip_1193.rs#L157-L158) is local only, so we'll need to make a workaround anyway.

@sergeyboyko0791 Could you please try to find a work around next sprint? It might be great to not reinvent the wheel, if work around itself is not too hard, of course 🙂